src/ui/reducers/timeline.ts

```diff
@@ -82,6 +84,46 @@ export const {
   paintsReceived,
 } = timelineSlice.actions;
 
+export const pointsReceivedThunk = (points: TimeStampedPoint[]): UIThunkAction => {
+  return (dispatch, getState) => {
+    const state = getState() as UIState;
+    dispatch(pointsReceived(points));
+    const focusRegion = getFocusRegion(state) as UnsafeFocusRegion | null;
+    if (!focusRegion) {
+      return;
+    }
+    // If we have just received points that we did not know about, those points
+    // might represent a better fit for the user-requested focus window than
+    // whatever points we were using before. If so, we will narrow the focus
+    // region down farther to the best-found points. However, in the case that
+    // we might be mucking about with time beyond the precision of floating
+    // point numbers, we should just chill, we are close enough to the user
+    // specified boundary (hence the epsilon check).
+    // See https://github.com/replayio/devtools/pull/7666 for more info.
```

Comment:
Why? Is it not true that we might filter out logs/events/network requests that are technically within the time range the user selected just because they're really close to a point we already knew about. I don't understand why this threshold is a good thing.

Reply:
I'm still kind of not understanding TBH